fdbclient/ManagementAPI.actor.cpp

```diff
@@ -465,6 +467,157 @@ bool isCompleteConfiguration(std::map<std::string, std::string> const& options)
 	       options.count(p + "storage_engine") == 1;
 }
 
+bool validTenantAndEncryptionAtRestMode(Optional<DatabaseConfiguration> oldConfiguration,
+                                        std::map<std::string, std::string> newConfig,
+                                        bool creating) {
+	if (creating) {
+		if (newConfig.count(encryptionAtRestModeConfKey.toString()) != 0) {
```

Comment:
nit: We need to add better documentation as following `if-else` is bit cumbersome and to some extent not clear it is satisfying all the conditions. 

We need to following following constraint:

**New cluster configuration**

1. If `encrypt_mode == domain_aware` **then** `tenant_mode == required`; any other configuration is a failure. Yes we can update `tenant_mode` from `optional/disabled -> required` when cluster is empty, but I think for now we can add it as a TODO and fix it later.
2.  If `encrypt_mode == cluster_aware/disabled` **then** `tenant_mode` can be anything.  

**Configuration update**
1. If `encrypt_mode == domain_aware` **then** `tenant_mode` update is a failure
2. If `encrypt_mode != tenant` **then** `tenant_mode` update is always allowed. 

Does the above comment covers all the usecase? If yes, we should simplify this block to cover all usecases, imo.

Reply:
Good point i can add some better documentation for this function. For the new cluster configuration all the cases you mentioned are being covered and i can simplify it a bit.

For the config update currently my if statement covers the following cases:
- if `encrypt_mode == domain_aware` then tenant_mode update is a failure unless the tenant_mode is `required` (yes generally noone will do this since its essentially changing the mode from required to required but in case its done by mistake then we should handle it)
- if `encrypt_mode == disabled` then we can simply allow any update to the tenant mode
- if `encrypt_mode == cluster_aware` then any tenant mode changes are allowed except going from optional|disabled -> required (which will fail right now). This is what the last if statement checks. We can always allow this (by default) but im not sure if that's what we want